tests/test_chat_templates.py

```diff
@@ -0,0 +1,151 @@
+}
+
+ANSWERS = {
+    ('Qwen/Qwen3-8B', 'thinking'): {
+        'single-turn': [[0, 12]],
+        'multi-turn': [[0, 22], [30, 42]],
+        'multi-turn-with-think': [[0, 22], [30, 42]],
+        'multi-turn-with-think-tools': [[0, 105], [113, 140]],
+    }
+}
+
+TAGS = {
+    'Qwen/Qwen3-8B': {
+        'asst_tag': '<|im_start|>assistant',
+        'end_tag': '<|im_end|>'
+    }
+}
+
+MAX_SEQ_LENGTH = 1024
+
+@pytest.mark.parametrize(
+    "tokenizer_name,chat_template_name,example_name", 
+    [
+        ('Qwen/Qwen3-8B', 'thinking', 'single-turn'),
+        ('Qwen/Qwen3-8B', 'thinking', 'multi-turn'),
+        ('Qwen/Qwen3-8B', 'thinking', 'multi-turn-with-think'),
+        ('Qwen/Qwen3-8B', 'thinking', 'multi-turn-with-think-tools'),
+    ]
+)
+def test_masking_strategy_span_search(
+    tokenizer_name: str,
+    chat_template_name: str,
+    example_name: List,
+    ignore_label: int = -100,
+):
+    tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+
+    # patch the chat template
+    tokenizer.chat_template = CHAT_TEMPLATE_EXAMPLES[chat_template_name]
+
+    tags = TAGS[tokenizer_name]
+
+    output = sft_span_seach_mask_out(
+        EXAMPLES[example_name],
+        tokenizer,
+        MAX_SEQ_LENGTH,
+        asst_tag=tags['asst_tag'],
+        end_tag=tags['end_tag'],
+        ignore_label=ignore_label,
+    )
+    input_ids = output[INPUT_IDS_KEY]
+    labels = output[LABELS_KEY]
+
+
+    spans = []
+    start = False
+    for i in range(len(labels)):
+        if labels[i] == ignore_label and not start:
+            spans.append([i])
+            start = True
+        elif labels[i] != ignore_label and start:
+            spans[-1].append(i)
+            start = False
+
```

Comment:
Thanks for adding tests!

Is this an independent test, though? Looks like repeated logic to me. Thought we'd have something like explicitly asserting that the elements of `labels` which should be masked, are actually masked. Might be misunderstanding the test, though.